xenserver/tasks.py

```diff
@@ -354,6 +358,18 @@ def _create_vm(session, vm, template, name, domain, ip, subnet, gateway, preseed
     if not network:
         raise NetworkError('Unable to locate VIF network')
 
+    extra_networks = []
+    for br in extra_network_bridges:
+        found = False
+        for net in session.xenapi.network.get_all():
```

Comment:
Is there a reason the `network.get_all()` call is not done once before looping through the user-provided bridges?